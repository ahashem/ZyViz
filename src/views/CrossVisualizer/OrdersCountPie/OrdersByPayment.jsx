import React from 'react';
import PropTypes from 'prop-types';

import PieChart from '../../../components/PieChart/PieChart';

/**
 * Abstract Pie Chart wrapper
 * @param data
 */
const OrdersByPayment = ({ data }) => {
  return (
    <PieChart
      data={data}
    />
  );
};

OrdersByPayment.propTypes = {
  data: PropTypes.arrayOf(PropTypes.shape({}))
};

OrdersByPayment.defaultProps = {
  data: [],
};

export default OrdersByPayment;
