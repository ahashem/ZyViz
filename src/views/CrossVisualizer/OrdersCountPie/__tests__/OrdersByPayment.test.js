import React from 'react';
import { shallow } from 'enzyme/build';
import OrdersByPayment from '../OrdersByPayment';

describe('<OrdersByPayment /> component', () => {
  it('should render with simple data', () => {
    const mockData = [
      { x: 'Cats', y: 35 },
      { x: 'Dogs', y: 40 },
      { x: 'Birds', y: 55 }
    ];

    const rendered = shallow(
      <OrdersByPayment
        data={mockData}
      />
    );

    expect(rendered).toMatchSnapshot();
  });
});

