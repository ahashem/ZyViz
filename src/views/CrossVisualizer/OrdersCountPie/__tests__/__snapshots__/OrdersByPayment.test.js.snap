// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrdersByPayment /> component should render with simple data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrdersByPayment
    data={
      Array [
        Object {
          "x": "Cats",
          "y": 35,
        },
        Object {
          "x": "Dogs",
          "y": 40,
        },
        Object {
          "x": "Birds",
          "y": 55,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "data": Array [
        Object {
          "x": "Cats",
          "y": 35,
        },
        Object {
          "x": "Dogs",
          "y": 40,
        },
        Object {
          "x": "Birds",
          "y": 55,
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "data": Array [
          Object {
            "x": "Cats",
            "y": 35,
          },
          Object {
            "x": "Dogs",
            "y": 40,
          },
          Object {
            "x": "Birds",
            "y": 55,
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
